src/application/controller/controller.c

```diff
@@ -1,2 +1,140 @@
 #include "application/controller/controller.h"
 
+QueueHandle_t internal_state_queue;
+QueueHandle_t output_queue;
+
+TickType_t timeout = pdMS_TO_TICKS(5);
+
+// internal: flight_condition_t
+typedef struct {
+    float dynamic_pressure;
+    float canard_coeff_lift;
+} flight_condition_t;
+
+// initialize structs: placeholder values
+static flight_condition_t flight_condition = {0};
+static controller_t controller_state = {0};
+static controller_input_t controller_input = {0};
+static controller_output_t controller_output = {0};
+
+/*
+Send `canard_angle`, the desired canard angle (radians) to CAN
+*/
+static w_status_t controller_send_can(float canard_angle) {
+    // Build the CAN msg using [canard-specific canlib function to be defined
+    // later]. Send this to can handler module’s tx
+    return W_SUCCESS;
+}
+
+/**
+ * Initialize controller module
+ * Must be called before RTOS scheduler starts
+ * @return W_SUCCESS if initialization successful
+ */
+w_status_t controller_init(void) {
+    w_status_t init_status = W_SUCCESS;
+
+    // Create internal state queue (length = 1)
+    internal_state_queue =
+        xQueueCreate(1, sizeof(controller_input_t)); // estimator function to controller task
+    output_queue = xQueueCreate(1, sizeof(controller_output_t));
+
+    // check queue creation
+    if (internal_state_queue == NULL || output_queue == NULL) {
+        init_status = W_FAILURE;
+        log_text("controller", "queue creation failed");
+    }
+
+    // initialize structs: non-placeholder values
+
+    // gain table initialization
+
+    // Log initialization status
+    if (init_status == W_SUCCESS) {
+        log_text("controller", "initialization successful");
+
+    } else {
+        log_text("controller", "initialization failed");
+    }
+
+    // return w_status_t state
+    return init_status;
+}
+
+/**
+ * Update controller with new state data - called by state estimation module
+ * @param new_state Latest state estimate from state estimation
+ * @return W_FAILURE if validation/queueing fails
+ */
+w_status_t controller_update_inputs(controller_input_t *new_state) {
+    if (xQueueOverwrite(internal_state_queue, new_state) == pdPASS) { // overwrite internal queue
```

Comment:
`xQueueOverwrite` can only ever return `pdPASS` so this error checking is ineffective
https://www.freertos.org/Documentation/02-Kernel/04-API-references/06-Queues/11-xQueueOverwrite